.github/workflows/lint-check.yml

```diff
@@ -0,0 +1,48 @@
+name: Static Analysis
+
+on:
+  pull_request:
+  push:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: 'ubuntu-latest'
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install tools
+        run: |
+          sudo apt-get update
+          sudo apt-get install cppcheck
+          pip install cmakelang
+
+      - name: Check cpp code formatting
+        run: |
+          git ls-files |\
+            grep -E '\.(c|cc|C|cii|cpp|cxx|c++|h|hii|hpp|hxx|h++)$' |\
+            xargs clang-format --dry-run --Werror
+
+      - name: Cppcheck
+        run: |
+          cppcheck --error-exitcode=1 ./src
+
+      - name: Check CMake code formatting
+        run: |
+          git ls-files |\
+            grep -E '(\.cmake$|CMakeLists.txt)' |\
+            xargs cmake-format --check
+
+      - name: Lint CMake code
+        # This step is configured to print all the output that cmake-lint would
+        # generate, but it will not cause a build job failure.
+        # If you need this step to fail when cmake-lint reports issues,
+        # simply remove the `|| true` from the last line.
+        run: |
+          git ls-files |\
+            grep -E '(\.cmake$|CMakeLists.txt)' |\
+            xargs cmake-lint || true
```

Comment:
I'm allowing failing `cmake-lint` checks not to cause a build failure, because some of the things it checks are frankly ridiculous. For instance, I get this warning on Sanitizers.cmake:

```
cmake/Sanitizers.cmake:02: [R0912] Too many branches 13/12
```

Excuse me? I'm not allowed to use more than 12 `if` statements? What's the justification for that?

I am aware that `cmake-lint` is configurable with a yaml file, or with CLI arguments, but I haven't done anything with that yet. There is an option to increase the max number of allowed `if` statements.

If you want to see the output from `cmake-lint`, here's a recent run: https://github.com/cpp-best-practices/cpp_starter_project/runs/3446661973?check_suite_focus=true#step:7:1. I think a lot of what it recommends is useful advice, particularly if we want this project to be an example of excellent CMake code that others should emulate/copy.

Reply:
I think this needs to be adjusted to use the static analyzers set up in CMake code, which already knows how they are meant to be run.

Reply:
File removed; this can be picked up in another PR